activities/files.go

```diff
@@ -184,10 +191,64 @@ func ListFiles(ctx context.Context, input FileInput) ([]paths.Path, error) {
 	}), err
 }
 
-func DeletePath(ctx context.Context, input FileInput) error {
+func DeletePath(ctx context.Context, input DeletePathInput) error {
 	log := activity.GetLogger(ctx)
 	activity.RecordHeartbeat(ctx, "DeletePath")
 	log.Info("Starting DeletePathActivity")
 
-	return os.RemoveAll(input.Path.Local())
+	if (input.Path.Path == "/") || (input.Path.Path == "") {
+		return merry.New("cannot delete root")
+	}
+
+	if input.RemoveAll {
+		return os.RemoveAll(input.Path.Local())
+	}
+
+	return os.Remove(input.Path.Local())
+}
+
+// WaitForFile waits until a file stops growing
+// Useful for waiting for a file to be fully uploaded, e.g. watch folders
+// Returns true if file is fully uploaded, false if failed, e.g. file doesnt exist after 5 minutes.
+func WaitForFile(ctx context.Context, input FileInput) (bool, error) {
+	logger := activity.GetLogger(ctx)
+	logger.Info("Starting WaitForFile")
+
+	// Use to cancel if file doesnt exist still after 5 minutes
+	startedAt := time.Now()
+
+	lastKnownSize := int64(0)
+	iterationsWhereSizeIsFreezed := 0
+
+	for {
+		res, err := os.Stat(input.Path.Local())
+		activity.RecordHeartbeat(ctx, res)
+		if err != nil {
+			fmt.Printf("Error: %v\n", err)
+			if time.Since(startedAt) > time.Minute*5 {
+				return false, err
+			}
+			time.Sleep(time.Second * 5)
+			continue
+		}
+		fmt.Printf("Stat res: %v\n", res)
+
+		if res.Size() < lastKnownSize {
+			return false, fmt.Errorf("file size decreased")
+		} else if res.Size() > lastKnownSize {
+			lastKnownSize = res.Size()
+			iterationsWhereSizeIsFreezed = 0
+			time.Sleep(time.Second * 5)
+			continue
+		}
+
+		iterationsWhereSizeIsFreezed++
+
+		fmt.Printf("Iterations: %v\n", iterationsWhereSizeIsFreezed)
+		if iterationsWhereSizeIsFreezed > 12 {
+			fmt.Printf("Returning\n")
+			return true, nil
+		}
+		time.Sleep(time.Second * 5)
```

Comment:
still some logging here. 

we don't store logs done to stdout, so if the logging is critical, use the temporal logger, or just remove it.